sdk/containerregistry/azure-containerregistry/README.md

```diff
@@ -69,74 +69,218 @@ For more information please see [Container Registry Concepts](https://docs.micro
 
 The following sections provide several code snippets covering some of the most common ACR Service tasks, including:
 
-- [List repositories](#list-repositories)
-- [List tags with anonymous access](#list-tags-with-anonymous-access)
-- [Set artifact properties](#set-artifact-properties)
-- [Delete images](#delete-images)
-
-Please note that each sample assumes there is a `CONTAINERREGISTRY_ENDPOINT` environment variable set to a string containing the `https://` prefix and the name of the login server, for example "https://myregistry.azurecr.io".
+- Registry operations:
+  - [List repositories](#list-repositories)
+  - [List tags with anonymous access](#list-tags-with-anonymous-access)
+  - [Set artifact properties](#set-artifact-properties)
+  - [Delete images](#delete-images)
+- Blob and manifest operations:
+  - [Upload images](#upload-images)
+  - [Download images](#download-images)
+  - [Delete manifest](#delete-manifest)
+  - [Delete blob](#delete-blob)
+
+Please note that each sample assumes there is a `CONTAINERREGISTRY_ENDPOINT` environment variable set to a string containing the `https://` prefix and the name of the login server, for example "https://myregistry.azurecr.io". Anonymous access samples are getting endpoint value from environment variable`CONTAINERREGISTRY_ANONREGISTRY_ENDPOINT`.
 
 ### List repositories
 
 Iterate through the collection of repositories in the registry.
 
-```python
-endpoint = os.environ["CONTAINERREGISTRY_ENDPOINT"]
+<!-- SNIPPET:sample_delete_tags.list_repository_names -->
 
-with ContainerRegistryClient(endpoint, DefaultAzureCredential(), audience="https://management.azure.com") as client:
+```python
+with ContainerRegistryClient(self.endpoint, self.credential) as client:
     # Iterate through all the repositories
     for repository_name in client.list_repository_names():
         print(repository_name)
 ```
 
+<!-- END SNIPPET -->
+
 ### List tags with anonymous access
 
 Iterate through the collection of tags in the repository with anonymous access.
 
-```python
-endpoint = os.environ["CONTAINERREGISTRY_ENDPOINT"]
+<!-- SNIPPET:sample_list_tags.list_tags_anonymous -->
 
-with ContainerRegistryClient(endpoint, DefaultAzureCredential(), audience="https://management.azure.com") as client:
-    manifest = client.get_manifest_properties("library/hello-world", "latest")
-    print(manifest.repository_name + ": ")
+```python
+endpoint = os.environ.get("CONTAINERREGISTRY_ANONREGISTRY_ENDPOINT")
+with ContainerRegistryClient(endpoint) as anon_client:
```

Comment:
Do we need to provide credential and audience? (I want to confirm if we need to update samples)

Reply:
This is anonymous access sample, credential is expected to be `None`.